skymp5-front/src/constructorComponents/chat/index.js

```diff
@@ -31,69 +65,129 @@ const Chat = (props) => {
       </span>
     );
   };
 
   const getList = () => {
     return props.messages.map((msg, index) => (
-            <div
-                className="msg"
-                key={`msg-${index}`}
-                style={{ marginLeft: '10px' }}
-            >
-              {getMessageSpans(msg)}
-            </div>
+      <div
+        className="msg"
+        key={`msg-${index}`}
+        style={{ marginLeft: '10px' }}
+      >
+        {getMessageSpans(msg)}
+      </div>
     ));
   };
 
+  const hideNonRPStyles = `
+  .nonrp {
+    display: none;
+  }
+  `;
+
   return (
-        <div id="chat">
-            <div className="list">{getList()}</div>
-            {isInputHidden
-              ? <></>
-              : <input
-                    id="chatInput"
-                    className={'show'}
-                    type="text"
-                    placeholder={placeholder !== undefined ? placeholder : ''}
-                    value={input}
-                    onChange={(e) => { updateInput(e.target.value); }}
-                    onFocus={(e) => changeInputFocus(true)}
-                    onBlur={(e) => changeInputFocus(false)}
-                    ref={inputRef}
-                />
-            }
-        </div>
+    <div id="chat">
+      <div className="chat-main">
+        <div className="list" ref={chatRef} onScroll={(e) => handleScroll()}>{getList()}</div>
+        {isInputHidden
+          ? <></>
+          : <div className='input'>
+            <div>
+              <input
+                id="chatInput"
+                className={'show'}
+                type="text"
+                placeholder={placeholder !== undefined ? placeholder : ''}
+                value={input}
+                onChange={(e) => { updateInput(e.target.value); }}
+                onFocus={(e) => changeInputFocus(true)}
+                onBlur={(e) => changeInputFocus(false)}
+                ref={inputRef}
+              />
+            </div>
+            <div className='chat-checkboxes'>
+              <ChatCheckbox id={'nonrp'} text={'non-rp'} isChecked={hideNonRP} onChange={(e) => changeNonRPHide(e.target.checked)} />
+              <ChatCheckbox id={'diceSound'} text={'dice sounds'} isChecked={disableDiceSounds} onChange={(e) => setDisableDiceSounds(e.target.checked)} />
+              {/* Maybe we will need it later: <ChatCheckbox id={'diceColor'} text={'dice colors'} isChecked={!disableDiceColors} onChange={(e) => setDisableDiceColors(!e.target.checked)} /> */}
+              <span className={`chat-message-limit ${input.length > MAX_LENGTH ? 'limit' : ''} text`}>{input.length}/{MAX_LENGTH}</span>
+            </div>
+          </div>
+        }
+      </div>
+      {
+        isInputHidden
+          ? <></>
+          : <Dices
+              isOpened={isPouchOpened}
+              setOpened={setPocuhOpened}
+              send={props.send}
+              disableSound={disableDiceSounds}
+          />
+      }
+      {
+        hideNonRP
```

Comment:
It's controversial, but I would implement this by adding the selector .nonrp to a wrapper. And this selector would be stored in a relative SCSS file.

Reply:
Fixed